File: EacademyApp.API/Models/Course.cs
using System.Collections.Generic;

namespace EacademyApp.API.Models
{
    public class Course
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public ICollection<CourseStudent> CourseStudents { get; set; }
        public ICollection<Module> Modules { get; set; }
        public Student Instructor { get; set; }
    }
}

File: EacademyApp-SPA/src/app/_models/course.ts
import { Student } from './student';
import { Module } from './module';
import { CourseStudents } from './courseStudents';

export interface Course {
    id: number;
    name: string;
    description: string;
    courseStudents?: CourseStudents[];
    modules?: Module[];
    instructor?: Student;
}
